cmd/config/docs/api-conventions/structured_output.md

```diff
@@ -0,0 +1,160 @@
+## Structured Output for Kubernetes CLI Tools
+Below is a protocol for emitting machine parseable structured output format (e.g. -o structured) from Kubernetes project tools. 
+This protocol would provide an API for better command integration & composition of tools in the Kubernetes ecosystem.
+
+## Structured Output Schema:
+Define the following schema for rendering structured output.
+
+```
+{
+  "$id": "https://example.com/person.schema.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "OutputMessage",
+  "version": "0.1.0"
+  "type": "object",
+	"description": "The specification for a structured output message",
+	"definitions": {
+		"error_details":{
+			"type": "object",
+			"required": ["error"],
+			"properties": {
+				"error": {
+					"type": "string",
+					"description": "The exception or error being logged e.g. 'User Error'
+					or ValidationException."
+				},
+				"context": {
+					"type": "string",
+					"description": "Any stack-trace or additional error context."
+				}
+			}
+		}
+	},
+	"required": ["version", "timestamp", "body"],
+	"properties": {
+		"version": {
+			"type": "string",
+			"description": "Semantic version of the message format e.g. v1.0.0.
+			Useful if message format changes before binary is updated."
+		},
+		"timestamp": {
+			"type": "string",
+			"description": "[RFC 3339](https://www.ietf.org/rfc/rfc3339.txt) encoded
+			timestamp."
+
+		},
+		"body": {
+			"type": "string",
+			"description": "The actual content of the message. JSON/YAML structured
+			content will be assumed to be resource content otherwise it will be
+			assumed to be plain text."
+		},
+		"error_detail": {
+			"$ref": "#/definitions/error_details",
+			"description": "Only present if the message is related to an error."
+
+		},
+		"verbosity": {
+			"enum": ["debug", "info", "warn", "error", "critical"]
+		}
+	}
+}
+```
+
+
+## `structued` output format
+
+For those tools (such as `kubectl`) that support the output flag (e.g.  `-o`, `--output`)
```

Comment:
Adding this to all existing kubectl commands will be a lot of work.  Perhaps adding it to future commands developed out of the kustomize repo for a start?

Reply:
This makes sense and I will note that